lib/trino-filesystem/src/main/java/io/trino/filesystem/cache/CacheFileSystem.java

```diff
@@ -79,6 +80,7 @@ public void deleteDirectory(Location location)
             throws IOException
     {
         delegate.deleteDirectory(location);
+        cache.expire(location);
```

Comment:
should expire happen before deleting directory or after?

Reply:
If you expire before, then another concurrent access can see the directory contents before deletion from remote storage. If you expire after, then another concurrent access can see the directory contents before deletion from the cache. The second one seems like cheaper option.

Reply:
if you expire before delete, then someone might populate the cache just before directory is deleted. In that case you end up with stalled cache data.